tests/prague/eip7692_eof_v1/eip7069_extcall/test_calls.py

```diff
@@ -318,6 +321,75 @@ def test_eof_calls_eof_mstore(
     )
 
 
+identity = Address(0x04)
+# `blake2f`` is chosen for the test because it fails unless args_size == 213, which is what we are
+# interested in.
+blake2f = Address(0x09)
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    [
+        Op.EXTCALL,
+        Op.EXTDELEGATECALL,
+        Op.EXTSTATICCALL,
+    ],
+)
+@pytest.mark.parametrize(
+    "precompile", [pytest.param(identity, id="identity"), pytest.param(blake2f, id="blake2f")]
+)
+def test_eof_calls_precompile(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    sender: EOA,
+    opcode: Op,
+    precompile: Address,
+):
+    """Test EOF contracts calling precompiles"""
+    env = Environment()
+
+    caller_contract = Container.Code(
+        Op.MSTORE(0, value_returndata_magic)
+        + Op.SSTORE(slot_call_result, opcode(address=precompile, args_offset=0, args_size=32))
+        + Op.SSTORE(slot_returndatasize, Op.RETURNDATASIZE)
+        + Op.SSTORE(slot_returndata, Op.RETURNDATALOAD(0))
+        + Op.SSTORE(slot_code_worked, value_code_worked)
+        + Op.STOP,
+    )
+    calling_contract_address = pre.deploy_contract(caller_contract)
+
+    tx = Transaction(
+        sender=sender,
+        to=Address(calling_contract_address),
+        gas_limit=5000000,
+        data="",
+    )
+
+    calling_storage = {
+        slot_code_worked: value_code_worked,
+        slot_call_result: value_eof_call_reverted
+        if opcode == Op.EXTDELEGATECALL
+        else value_eof_call_failed
+        if precompile == blake2f
+        else value_eof_call_worked,
+        slot_returndatasize: 0 if opcode == Op.EXTDELEGATECALL or precompile == blake2f else 32,
+        slot_returndata: 0
+        if opcode == Op.EXTDELEGATECALL or precompile == blake2f
+        else value_returndata_magic,
```

Comment:
For clarity I think explicit values in the params would work better.  Instead of relying on the multiple params for a cartesian join just list all 6 combos in one joint params block and make the slot values parameters as well.\, like you did in `test_callee_fails`.

Reply:
yes, now I look at it after a day it doesn't make any sense :joy: and the formatting doesn't help. Changed.